core/sdk.go

```diff
@@ -35,66 +37,188 @@ type PolicyResult struct {
 // Warning: This interface is experimental, and it could change with breaking also in rond patches.
 // Does not use outside this repository until it is not ready.
 type SDK interface {
-	// Warning: this method will be removed in the near future. Do not use it outside Rond.
-	Metrics() metrics.Metrics
-
 	FindEvaluator(logger *logrus.Entry, method, path string) (SDKEvaluator, error)
-	EvaluatorFromConfig(logger *logrus.Entry, config openapi.RondConfig) SDKEvaluator
 }
 
 // Warning: This interface is experimental, and it could change with breaking also in rond patches.
 // Do not use outside this repository until it is not ready.
 type SDKEvaluator interface {
 	Config() openapi.RondConfig
-	PartialResultsEvaluators() PartialResultsEvaluators
+
+	EvaluateRequestPolicy(ctx context.Context, req RondInput, userInfo types.User) (PolicyResult, error)
+	EvaluateResponsePolicy(ctx context.Context, rondInput RondInput, userInfo types.User, decodedBody any) ([]byte, error)
 }
 
 type evaluator struct {
 	rond       rondImpl
 	logger     *logrus.Entry
 	rondConfig openapi.RondConfig
+
+	routeInfo openapi.RouterInfo
+}
+
+func (e evaluator) metrics() metrics.Metrics {
+	return e.rond.metrics
+}
+
+func (e evaluator) partialResultEvaluators() PartialResultsEvaluators {
+	return e.rond.partialResultEvaluators
 }
 
 func (e evaluator) Config() openapi.RondConfig {
 	return e.rondConfig
 }
 
-func (e evaluator) PartialResultsEvaluators() PartialResultsEvaluators {
-	return e.rond.evaluator
+func (e evaluator) EvaluateRequestPolicy(ctx context.Context, req RondInput, userInfo types.User) (PolicyResult, error) {
+	if req == nil {
+		return PolicyResult{}, fmt.Errorf("RondInput cannot be empty")
+	}
+
+	rondConfig := e.Config()
+
+	input, err := req.FromRequestInfo(userInfo, nil)
+	if err != nil {
+		return PolicyResult{}, err
+	}
+
+	regoInput, err := CreateRegoQueryInput(e.logger, input, RegoInputOptions{
+		EnableResourcePermissionsMapOptimization: rondConfig.Options.EnableResourcePermissionsMapOptimization,
+	})
+	if err != nil {
+		return PolicyResult{}, nil
+	}
+
+	var evaluatorAllowPolicy *OPAEvaluator
+	if !rondConfig.RequestFlow.GenerateQuery {
+		evaluatorAllowPolicy, err = e.partialResultEvaluators().GetEvaluatorFromPolicy(ctx, rondConfig.RequestFlow.PolicyName, regoInput, e.rond.evaluatorOptions)
+		if err != nil {
+			return PolicyResult{}, err
+		}
+	} else {
+		evaluatorAllowPolicy, err = e.rond.opaModuleConfig.CreateQueryEvaluator(ctx, e.logger, rondConfig.RequestFlow.PolicyName, regoInput, e.rond.evaluatorOptions)
+		if err != nil {
+			return PolicyResult{}, err
+		}
+	}
+
+	opaEvaluationTimeStart := time.Now()
+
+	_, query, err := evaluatorAllowPolicy.PolicyEvaluation(e.logger, &rondConfig)
```

Comment:
why are you ignoring the first return value? isn't this the `allowed` boolean?

I know it was ignored before as well but I don't remember why 😢

Reply:
The first value is the responseBody. Since in response we use directly the Evaluate function, the first params is useless and we could remove it from the interface. What do you think?

Reply:
Yes, it makes sense! Maybe we could do that in another pr

Reply:
Sure, I add an issue